src/Tribe/Admin/Move_Tickets.php

```diff
@@ -0,0 +1,640 @@
+		exit();
+	}
+
+	/**
+	 * Returns a list of post types for which tickets are currently enabled.
+	 *
+	 * The list is expressed as an array in the following format:
+	 *
+	 *     [ 'slug' => 'name', ... ]
+	 *
+	 * @return array
+	 */
+	protected function get_post_types_list() {
+		$types_list = array( 'all' => __( 'All supported types', 'tribe-tickets' ) );
+
+		foreach ( Tribe__Tickets__Main::instance()->post_types() as $type ) {
+			$pto = get_post_type_object( $type );
+			$types_list[ $type ] = $pto->label;
+		}
+
+		return $types_list;
+	}
+
+	/**
+	 * Responds to requests for a list of possible destination posts.
+	 */
+	public function get_post_choices() {
+		if ( ! wp_verify_nonce( $_POST['check' ], 'move_tickets' ) ) {
+			wp_send_json_error();
+		}
+
+		$args = wp_parse_args( $_POST, array(
+			'post_type'    => '',
+			'search_terms' => '',
+			'ignore'       => '',
+		) );
+
+		wp_send_json_success( array( 'posts' =>  $this->get_possible_matches( $args ) ) );
+		exit();
+	}
+
+	/**
+	 * Returns a list of posts that could be possible homes for a ticket
+	 * type, given the constraints in optional array $request (if not set,
+	 * looks in $_POST for the corresponding values):
+	 *
+	 * - 'post_type': string or array of post types
+	 * - 'search_term': string used for searching posts to narrow the field
+	 *
+	 * @param array|null $request post parameters (or looks at $_POST if not set)
+	 *
+	 * @return array
+	 */
+	protected function get_possible_matches( array $request = null ) {
+		// Take the params from $request if set, else look at $_POST
+		$params = wp_parse_args( is_null( $request ) ? $_POST : $request, array(
+			'post_type' => array(),
+			'search_terms' => '',
+			'ignore' => '',
+		) );
+
+		// The post_type argument should be an array (of all possible types, if not specified)
+		$post_types = array_filter( (array) $params[ 'post_type' ] );
+
+		if ( empty( $post_types ) || 'all' === $params[ 'post_type' ] ) {
+			$post_types = array_keys( $this->get_post_types_list() );
+		}
+
+		/**
+		 * Controls the number of posts returned when searching for posts that
+		 * can serve as ticket hosts.
+		 *
+		 * @param int $limit
+		 */
+		$limit = (int) apply_filters( 'tribe_tickets_find_ticket_type_host_posts_limit', 100 );
+
+		$ignore_ids = is_numeric( $params[ 'ignore' ] ) ? array( absint( $params[ 'ignore' ] ) ) : array();
+
+		return $this->format_post_list( get_posts( array(
+			'post_type'      => $post_types,
+			'posts_per_page' => $limit,
+			'eventDisplay'   => 'custom',
+			'orderby'        => 'title',
+			'order'          => 'ASC',
+			's'              => $params[ 'search_terms' ],
+			'post__not_in'   => $ignore_ids,
+		) ) );
+	}
+
+	/**
+	 * Given an array of WP_Post objects, returns an array containing the post title
+	 * of each (with the post ID as the index).
+	 *
+	 * @param array $query_results
+	 *
+	 * @return array
+	 */
+	protected function format_post_list( array $query_results ) {
```

Comment:
This appears identical to [this method](https://github.com/moderntribe/event-tickets/pull/229/files#diff-41c69c826caa255771804f564897194fR135).